snapcraft/internal/repo.py

```diff
@@ -83,7 +83,8 @@ def install_build_packages(packages):
     with apt.Cache() as apt_cache:
         for pkg in unique_packages:
             try:
-                if not apt_cache[pkg].installed:
+                pkg_name, _ = _get_pkg_name_parts(pkg)
+                if not apt_cache[pkg_name].installed:
```

Comment:
I'm working on an update to check if the installed version matches the requested version and install the requested version if they don't match.  I'm assuming apt will do the right thing here.